src/components/utilities.js

```diff
@@ -150,3 +152,58 @@ export function getResultsNotFoundTranslationKey(searchType, hasQuery) {
+
+  return result ? result.join('') : '';
+}
+
+export function formatCustomLabelsValues(formValues) {
+  const displayLabels = getDisplayLabels(formValues);
+  const formattedFormValues = Object.keys(displayLabels).map((label) => {
+    const id = label[0];
+
+    return {
+      type: 'customLabel',
+      attributes: {
+        displayOnFullTextFinder: formValues[`${id}displayOnFullTextFinder`] || false,
+        displayOnPublicationFinder: formValues[`${id}displayOnPublicationFinder`] || false,
+        displayLabel: displayLabels[label],
+        id,
+      },
+    };
+  });
+
+  return formattedFormValues;
+}
+
+export function getAllCustomLabelsFields(customLabels) {
+  // Array [1, 2, 3, 4, 5] here just like a helper
+  // On settings custom labels page user should see 5 rows of custom labels inputs
+  // But sometimes we could get from backend less then 5 custom labels
+  // And the missing custom labels should be present too in correct order
+  const allCustomLabels = [1, 2, 3, 4, 5].map(id => {
```

Comment:
I don't think that it's a nice idea to pretend that we have a full set of custom labels and put them in an array. We always have 5 static fields on the page, it doesn't depend on the backend response. I would better formatted the initial values like this: 
```js
custom-fields: {
   1: {cfData}
   4: {cfData}
   5: {cfData}
}
```
and then render all five Fields, although there is only 3 in the database:
```js
<Field name="custom-fields.1" />
<Field name="custom-fields.2" />
<Field name="custom-fields.3" />
<Field name="custom-fields.4" />
<Field name="custom-fields.5" />
```
FinalForm will leave the fields which are not present in it's state empty